062-support-tiered-storage.md

```diff
@@ -0,0 +1,126 @@
+
+The Strimzi Cluster Operator will automatically configure any required configurations for the RLMM such as the basic thread pool configs required to create the internal Kafka client and talk to the brokers.
+
+Any additional RLMM configuration options such as performance tuning or security config options can be put in `.spec.kafka.config` directly using the `rlmm.config.` prefix.
+
+Example can be found below.
+
+### Example configuration
+
+The below config define a sample configuration for tiered storage setup, using a `custom` RSM type.
+```
+kafka:
+  tieredStorage:
+    type: customRemoteStorageManager
+    className: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+    classPath: /opt/kafka/plugins/tiered-storage-s3/*
+    config:
+    # A map with String keys and String values. Keys will be automatically prefixed with `rsm.config.`.
+      storage.bucket.name: my-bucket
+  config:
+    ...
+    rlmm.config.remote.log.metadata.common.client.bootstrap.servers: broker-bootstrap.com:9091
+    rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+    rlmm.config.remote.log.metadata.topic.replication.factor: 1
+```
+
+The configuration above will get translated to the below Kafka broker config:
+```
+# RLMM configuration generated by Strimzi
+remote.log.storage.system.enable: true
+remote.log.metadata.manager.impl.prefix: rlmm.config.
+remote.log.metadata.manager.class.name: org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager
+#RSM configs set by the operator and by the user
+remote.log.storage.manager.class.name: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+remote.log.storage.manager.class.path: /opt/kafka/plugins/tiered-storage-s3/*
+remote.log.storage.manager.impl.prefix: rsm.config.
+rsm.config.storage.bucket.name=my-bucket
+# # Custom RLMM configs
+rlmm.config.remote.log.metadata.common.client.bootstrap.servers: broker-bootstrap.com:9091
+rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+rlmm.config.remote.log.metadata.topic.replication.factor: 1
+```
+
+### Validation
+
+There are certain [limitations](https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Tiered+Storage+Early+Access+Release+Notes) for tiered storage enabled cluster.
+
+We choose not to validate the error conditions at Strimzi operator layer because:
+
+* At least some of them are expected to be removed in the future
+* Validating them might be quite complex (for example, IIRC, the compacted topics can be set in the topic itself as well as in the broker defaults)
+* Validating this in UTO would capture only a part of the topics and not prevent the users from misconfiguration. So they will still need to be aware of these rules and learn to respect them.
```

Comment:
I guess these rules would be covered in supporting Strimzi documentation? Do we mention?

Reply:
Hi Paul, thank you for your comment, I am not sure if I get this question, to be clear these are not rules but just justifications on why we not want to do exhaustive validations for those limitations.